t/loop-removal.t

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env perl
+use strict;
+use warnings;
+
+no indirect qw(fatal);
+use utf8;
+
+use Test::More;
+use Test::NoLeaks;
+use Future::AsyncAwait;
+use Syntax::Keyword::Try;
+
+use IO::Async::Loop;
+use Database::Async;
+use Database::Async::Engine::PostgreSQL;
+
+my $loop = IO::Async::Loop->new;
+
+my $uri = URI->new('postgresql://example@127.0.0.1:5000/empty?password=example-password');
```

Comment:
A question: why put `password` as a URI parameter? I tried but seems URI::db doesn't support such format. password need to put after user: 
https://metacpan.org/pod/URI::db#Format

db:engine:[//[user[:password]@][host][:port]/][dbname][?params][#fragment]
--


</div></div></pre><dl style="box-sizing: border-box; scroll-margin-top: 60px; margin-top: 0px; margin-bottom: 21px;"><dt id="db" style="box-sizing: border-box; scroll-margin-top: 60px; line-height: 1.42857; font-weight: bold; font-size: 1.1em;"><a href="https://metacpan.org/pod/URI::db#db" class="anchor" style="box-sizing: border-box; scroll-margin-top: 60px; background-color: rgb(255, 255, 255); color: rgb(51, 122, 183); text-decoration: none; padding-left: 9px; width: 26px; margin-left: -26px; vertical-align: top; font-size: 14px; display: inline-block; font-family: &quot;Helvetica Neue&quot;, Helvetica, Arial, sans-serif; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 700; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px;"><span class="fa fa-bookmark black" style="box-sizing: border-box; scroll-margin-top: 60px; -webkit-font-smoothing: antialiased; display: inline-block; font-style: normal; font-variant: normal; text-rendering: auto; line-height: 1; font-family: &quot;Font Awesome 5 Free&quot;; font-weight: 900; color: rgb(0, 0, 0); visibility: hidden;"></span></a><code style="box-sizing: border-box; scroll-margin-top: 60px; font-family: Menlo, Monaco, Consolas, &quot;Liberation Mono&quot;, &quot;Bitstream Vera Sans Mono&quot;, Courier, &quot;Courier New&quot;, monospace; font-size: 14.85px; padding: 2px 4px; color: rgb(51, 51, 51); background-color: rgb(245, 245, 245); border-radius: 4px; white-space: normal; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 700; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><br class="Apple-interchange-newline">


db:engine:[//[user[:password]@][host][:port]/][dbname][?params][#fragment]
--




db:engine:[//[user[:password]@][host][:port]/][dbname][?params][#fragment]
--




db:engine:[//[user[:password]@][host][:port]/][dbname][?params][#fragment]
--




db:engine:[//[user[:password]@][host][:port]/][dbname][?params][#fragment]
--